src/pages/DoctorsPage.tsx

```diff
@@ -1,7 +1,72 @@
+import { Card } from "../components/shared/card/Card";
+import { useEffect, useState } from "react";
+import { Button } from "../components/shared/button/Button";
+import { useNavigate } from "react-router-dom";
+import { fetchDoctors } from "../components/services/patientService";
+import { specialityData, specialityImages } from "../assets/assets";
+import useSWR from "swr";
+
 export function DoctorsPage() {
+  const [filterDoc, setFilterDoc] = useState([]);
+  const [selectedSpeciality, setSelectedSpeciality] = useState(null);
+  const navigate = useNavigate();
+
+  const { data: doctors, error, isLoading } = useSWR("doctors", fetchDoctors);
+
+  if (isLoading) return <div>Loading doctors...</div>;
+  if (error) return <div>Error loading doctors: {error.message}</div>;
+
+  const applyFilter = () => {
+    if (selectedSpeciality === "All Doctors" || !selectedSpeciality) {
+      setFilterDoc(doctors); // Jeśli "Wszyscy lekarze", nie filtrujemy
+    } else {
+      setFilterDoc(
+        doctors.filter((doc) => doc.speciality === selectedSpeciality)
+      );
+    }
+  };
+
+  useEffect(() => {
+    applyFilter(); // Wywołuje filtr za każdym razem, gdy zmieni się `selectedSpeciality`
+  }, [selectedSpeciality, doctors]); // Uruchamia się, gdy `selectedSpeciality` zostanie zaktualizowane
+
   return (
     <>
-      <div>Doctors</div>
+      <div className="flex gap-8 p-4 max-w-screen-lg mx-auto">
+        <div className="flex flex-col gap-2 w-1/4">
+          {specialityData.map((doc, index) => (
+            <p
+              className={`px-3 py-2 border cursor-pointer border-gray-300 rounded-md text-sm shadow-sm hover:bg-gray-100 transition-all duration-300 ${
+                selectedSpeciality === doc.speciality ? "bg-gray-200" : ""
+              }`}
+              onClick={() => setSelectedSpeciality(doc.speciality)}
+              key={index}
```

Comment:
https://medium.com/react-in-the-real-world/why-you-should-never-ever-use-indices-as-react-keys-8329472170be

Nie używamy indeksu jako klucza niestety. Jest to unikalny identyfikator który pozwala Reactowi wynioskować który rekord musi zaktualizowac podczas zmian. Gdy usuniemy jakies unikalne ID np (1dj1dce) to IDki pozostałych elementów pozostaną niezmienne, wiec React jedynie usunie jeden element listy.
Jezeli dasz index, to przy usuwaniu zmieni sie length tablicy i każdy user po tym usuniętym otrzyma nowe ID, co wyrenderuje na nowo ich mimo ze nic sie nie zmieni.